src/components/__tests__/publicAPI.spec.js

```diff
@@ -0,0 +1,177 @@
+    // and also specify the `isRequired` field so PropTypes.string.isRequired
+    // works as intended
+    /* eslint-disable no-restricted-syntax */
+    for (const type of primitive) {
+      Object.defineProperty(PropTypes, type, {
+        get() {
+          const value = {
+            type,
+          };
+          Object.defineProperty(value, 'isRequired', {
+            enumerable: false,
+            get() {
+              return {
+                type,
+                isRequired: true,
+              };
+            },
+          });
+          return value;
+        },
+      });
+    }
+
+    // For complex/chainable prop types, we'll redefine the value to a function
+    // and have it return the prop type information as a result of calling the
+    // prop-type. We'll also need to define the `isRequired` field similar to
+    // how we used it in the primitive prop types
+    for (const type of complex) {
+      // eslint-disable-next-line func-names
+      PropTypes[type] = function(...args) {
+        const value = {
+          type,
+          args,
+        };
+
+        Object.defineProperty(value, 'isRequired', {
+          enumerable: false,
+          get() {
+            return {
+              type,
+              args,
+              isRequired: true,
+            };
+          },
+        });
+
+        return value;
+      };
+    }
+    /* eslint-enable no-restricted-syntax */
+
+    return PropTypes;
+  });
+});
+
+/**
+ * We construct our public API by iterating through our entrypoint and finding
+ * the components that we export and their corresponding API.
+ */
+describe('PublicAPI', () => {
+  test('should only change with a semver change', () => {
+    function mapComponentToAPI(Component) {
+      const api = {};
+
+      Object.keys(Component).forEach(key => {
+        // There are a couple of properties on components that we don't believe
+        // are part of our API, such `_` prefixed variables, or capture details
+        // that are internal to a library-specific piece of functionality. For
+        // example, React's context functionality.
+        if (key[0] === '_') {
+          return;
+        }
+
+        if (key === 'Consumer') {
+          api[key] = 'React.Consumer';
+          return;
+        }
+
+        if (key === 'Provider') {
+          api[key] = 'React.Provider';
+          return;
+        }
+
+        // Handle components defined as fields on a component, for example
+        // `MultiSelect.Filterable`
+        if (
+          typeof Component[key] === 'function' &&
+          key !== 'render' &&
+          isValidElementType(Component[key])
+        ) {
+          api[key] = mapComponentToAPI(Component[key]);
+          return;
+        }
+
+        api[key] = Component[key];
+      });
+
+      return api;
+    }
+
+    // eslint-disable-next-line global-require
+    const CarbonAddonsIoTReact = require('../../index');
```

Comment:
I'm wondering if we can move this import to the top of this file, similar to how we handle other imports? With that, we should also be able to avoid using a global require - For example:

```js
import * as Components from '../../index';
```

Reply:
For sure, I understand this — as mentioned in the PR desc, the test is copied exactly as-is.